website/server/controllers/api-v3/tasks.js

```diff
@@ -653,6 +656,49 @@ api.scoreTask = {
       user.save(),
       task.save(),
     ];
+
+    // Handle shared completion
+    if (task.type === 'todo' && direction === 'up' && task.group && task.group.sharedCompletion === 'singleCompletion') {
+      let masterTask = await Tasks.Task.findOne({
+        _id: task.group.taskId,
+      }).exec();
+
+      masterTask.completed = true;
+      promises.push(masterTask.save());
+
+      const tasksToRemove = await Tasks.Task.find({
```

Comment:
can this be a remove mongodb operation directly instead of a find + remove for each individual task?